File: 2.js
function fib(range) {
  let fibonacci = [1, 2];

  while (true) {
    const nextNumber =
      fibonacci[fibonacci.length - 1] + fibonacci[fibonacci.length - 2];
    if (nextNumber >= range) break;
    else fibonacci.push(nextNumber);
  }

  return fibonacci;
}

function filterEven(numbers) {
  return numbers.filter((number) => number % 2 === 0);
}

function sum(numbers) {
  return numbers.reduce((sum, number) => (sum += number), 0);
}

function main() {
  return sum(filterEven(fib(4000000)));
}

module.exports = main;

File: 2.clj
(defn fibonacci [till] 
    (loop [fibs '(2 1)]
        (do
            (def next-fib (+ (first fibs) (second fibs)))
            (if (>= next-fib till) 
                fibs
                (recur (conj fibs next-fib))
            )
        )
    )
)

(defn sum [list]
    (loop [sum 0 lst list]
        (if (empty? lst)
            sum
            (recur (+ sum (first lst)) (rest lst) )
        )
    )
)

(println (sum (filter even? (fibonacci 4000000))))


